mgexpose/downstream.py

```diff
@@ -0,0 +1,488 @@
+#!/usr/bin/env python
+# pylint: disable=R0912,R0914,R0915
+import os
+
+from collections import Counter, defaultdict
+
+from .gffio import read_mge_genomic_islands_gff
+
+from .base_logger import logger
+
+def stat_nested(islands):
+    """Calculate statistics on nested MGEs and return as a dictionary."""
+    total = 0
+    nested = 0
+    for island in islands:
+        if island.mge_type == "nested":
+            nested += 1
+        total += 1
+    nested_percentage = (nested / total * 100) if total > 0 else 0
+    non_nested_percentage = ((total - nested) / total * 100) if total > 0 else 0
```

Comment:
```suggestion
    total = len(islands)
    nested = sum(island.mge_type == "nested" for island in islands)
    nested_percentage, non_nested_percentage = 0, 0
    if total:
        fr_nested = nested / total
        fr_non_nested = 1 - fr_nested
        nested_percentage, non_nested_percentage = 100 * fr_nested, 100 * fr_non_nested
```

- total is number of islands
- no need for if block in for loop, just sum up the number of nested islands
- no need to check total > 0 and to calculate the fraction / percentage twice